src/Backend/Lua/Builtin.hs

```diff
@@ -118,30 +118,62 @@ builtins =
           return var[1]
         end
       |] )
-  , ( vStrVal, "__builtin_strval", [], Nothing
+  , ( vStrVal, "__builtin_strval", [], Just (1, \[l] -> (mempty, [ [lua|%l|] ]))
     , [luaStmts|
         local function __builtin_strval(x)
           return x
         end
       |] )
-  , ( vKSTR, "__builtin_kstr", [], Nothing
+  , ( vKSTR, "__builtin_kstr", [], Just (1, \[l] -> (mempty, [ [lua|%l|] ]))
     , [luaStmts|
         local function __builtin_kstr(x)
           return x
         end
       |] )
-  , ( vIntVal, "__builtin_intval", [], Nothing
+  , ( vIntVal, "__builtin_intval", [], Just (1, \[l] -> (mempty, [ [lua|%l|] ]))
     , [luaStmts|
         local function __builtin_intval(x)
           return x
         end
       |] )
-  , ( vKINT, "__builtin_kint", [], Nothing
+  , ( vKINT, "__builtin_kint", [], Just (1, \[l] -> (mempty, [ [lua|%l|] ]))
     , [luaStmts|
         local function __builtin_kint(x)
           return x
         end
       |] )
+  , ( vROWCONS, "__builtin_rowcons", [], Just (1, \[l] -> (mempty, [ [lua|%l|] ]))
+    , [luaStmts|
+        local function __builtin_rowcons(x)
+          return x
+        end
+      |] )
+  , ( vExtend, "__builtin_extend", [], Nothing
+    , [luaStmts|
+        local function __builtin_extend(key, value, record)
```

Comment:
I wonder if it's worth having a generic `__builtin_copy` function, which both extend and restrict depend on? I could also get the codegen to emit a call to it, instead of inlining the loop.